src/bootstrap/builder.rs

```diff
@@ -360,10 +367,19 @@ impl StepDescription {
         for (desc, should_run) in v.iter().zip(&should_runs) {
             let pathsets = should_run.pathset_for_paths_removing_matches(&mut paths, desc.kind);
             if !pathsets.is_empty() {
-                desc.maybe_run(builder, pathsets);
+                hosts_check.push(desc.maybe_run(builder, pathsets));
             }
         }
 
+        // sanity checks on hosts
+        if !hosts_check.contains(&Some(())) {
```

Comment:
This seems like hosts_check is really something like a bool, i.e., we don't care about the length etc. Can we store it that way?

Reply:
A bool would solve it. Obviously I didn't note this down anywhere, but I thought I would fix this and then talk to someone about creating an issue to solve what kind of error message we should get if we had no succesful runs due to the exclusion test at [line 288](https://github.com/rust-lang/rust/blob/0118a8b34580ed7b34d7f8071a6a78b97f38ab65/src/bootstrap/builder.rs#L288). I thought it would be nice to return an Option storing a `ReasonForNotRunning` enum. 

Let me change this to a bool for now and expand to the vector of options if someone could greenlight the other issue. I will be at a computer with a compiler tomorrow so the change will come then :)